upt_macports/upt_macports.py

```diff
@@ -13,7 +15,42 @@ def __init__(self):
     def create_package(self, upt_pkg, output):
         self.upt_pkg = upt_pkg
         self.logger.info(f'Hello, creating the package')
-        print(self._render_makefile_template())
+        if output is None:
+            print(self._render_makefile_template())
+        else:
+            self._create_output_directories(upt_pkg, output)
+            self._create_portfile()
+
+    def _create_output_directories(self, upt_pkg, output_dir):
+        """Creates the directory layout required"""
+        self.logger.info(f'Creating the directory structure in {output_dir}')
+        upt2macports = {
+            'cpan': 'perl',
+            'pypi': 'python',
+            'rubygems': 'ruby',
+        }
+        try:
+            port_category = upt2macports[self.upt_pkg.frontend]
+        except KeyError:
+            sys.exit(
+                f'Unknown port category for frontend {self.upt_pkg.frontend}')
+        folder_name = self._normalized_macports_folder(upt_pkg.name)
+        self.output_dir = os.path.join(
+            output_dir, port_category, folder_name)
+        try:
+            os.makedirs(os.path.join(self.output_dir), exist_ok=True)
```

Comment:
os.path.join with a single argument seems a bit useless here, isn't it?

Reply:
true ;) Besides that, I think this is ready! I'll do some final testing tomorrow morning, but if nothing pops up there, any concerns with me merging this PR?